ucsb-cs156/proj-happycows
# Pc reports page

# Overview

In this PR, we create a reports page where the Admin can view reports created by the InstructorReport job.

# Merge these PRs first

* #58 (report jobs) 
* #59 (report controllers)
* #60 (springfox -> springdoc)
* #61 (report components)

# Issues Addressed

Closes #18 

# Link to Storybook

TODO link to storybook

# Details

This PR finishes work on the first draft of creating instructor reports. 

- [x] There is a backend endpoint where the instructor can request all reports.
- [x] The instructor can request an instructor report on the jobs page, for all commons, or a specific commons
- [x] There is a job that *could* be scheduled (though it may not be scheduled yet) to run these reports periodically (e.g. once a day, twice a day, etc.)
- [x] The instructor can view all reports that have been generated on the Admin / Reports page at `/admin/reports`
- [x] There is a link on the Navbar Admin dropdown for Reports that takes the user to `/admin/reports`
- [x] The instructor can select a report and view the report detail at `/admin/reports/:report_id`; that page uses the `ReportHeaderTable` and `ReportDetailTable` to show the results.

# Test Plan (for interactive testing)

1.  Create a user commons or two.  Have a low capacity so that health will go down with relatively few cows.
2. Have at least two users join one of the user commons
3. Generate a report
4. Have users buy some cows (enough to be over capacity), and run the Milk The Cows and or Cow Health update jobs a few times, alternating with generating reports.
5. View the reports page and see if the reports reflect the changes over time.
